pkg/transform/kubernetes/kubernetes.go

```diff
@@ -189,14 +190,91 @@ func (t *Transformer) CreateDeployments(o *object.Service) ([]runtime.Object, er
 			})
 		}
 
+		// TODO: It is assumed that the check is done about the existence of volume in root level volume section
+		for _, mount := range c.Mounts {
+
+			volumeMount := api_v1.VolumeMount{
+				Name:      mount.VolumeName,
+				ReadOnly:  mount.ReadOnly,
+				MountPath: mount.MountPath,
+				SubPath:   mount.VolumeSubPath,
+			}
+
+			kc.VolumeMounts = append(kc.VolumeMounts, volumeMount)
+
+			// if this mount does not exist in emptydir then this is coming from root level volumes directive
+			// if tomorrow we add support for ConfigMaps or Secrets mounted as volumes the check should be done
+			// here to see if it is not coming from configMaps or Secrets
+			if !o.EmptyDirVolumeExists(mount.VolumeName) {
+				volume := api_v1.Volume{
+					Name: mount.VolumeName,
+					VolumeSource: api_v1.VolumeSource{
+						PersistentVolumeClaim: &api_v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: mount.VolumeName,
+						},
+					},
+				}
+				d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes, volume)
+			}
+		}
+
 		d.Spec.Template.Spec.Containers = append(d.Spec.Template.Spec.Containers, kc)
 	}
 
+	// make entry of emptydir in deployment volume directive
+	for _, emptyDir := range o.EmptyDirVolumes {
+		volume := api_v1.Volume{
+			Name: emptyDir.Name,
+			VolumeSource: api_v1.VolumeSource{
+				EmptyDir: &api_v1.EmptyDirVolumeSource{},
+			},
+		}
+		d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes, volume)
+	}
+
 	result = append(result, d)
 
 	return result, nil
 }
 
+// Create Kubernetes Persistent Volume Claim
+func (t *Transformer) CreatePVC(volume object.Volume) (runtime.Object, error) {
+
+	size, err := resource.ParseQuantity(volume.Size)
+	if err != nil {
+		return nil, err
+	}
+
+	pvc := &api_v1.PersistentVolumeClaim{
+		ObjectMeta: api_v1.ObjectMeta{
+			Name: volume.Name,
+		},
+		Spec: api_v1.PersistentVolumeClaimSpec{
+			Resources: api_v1.ResourceRequirements{
+				Requests: api_v1.ResourceList{
+					api_v1.ResourceStorage: size,
+				},
+			},
+		},
+	}
+
+	switch volume.AccessMode {
+	case "ReadWriteOnce":
+		pvc.Spec.AccessModes = []api_v1.PersistentVolumeAccessMode{api_v1.ReadWriteOnce}
+	case "ReadOnlyMany":
+		pvc.Spec.AccessModes = []api_v1.PersistentVolumeAccessMode{api_v1.ReadOnlyMany}
+	case "ReadWriteMany":
+		pvc.Spec.AccessModes = []api_v1.PersistentVolumeAccessMode{api_v1.ReadWriteMany}
+	}
```

Comment:
Can we add `default` section that  will cause error?
I know that it shouldn't happen because its validated before, but you never know ;-) That one line might save a lot of debugging in future ;-)